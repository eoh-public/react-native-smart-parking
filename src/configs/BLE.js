export default {
  BLE_REMOTE_SERVICE_UUID: '4fafc201-1fb5-459e-8fcc-c5c9c331914b',
  BLE_REMOTE_CHARACTERISTIC_UUID: 'beb5483e-36e1-4688-b7f5-ea07361b26a8',
};
