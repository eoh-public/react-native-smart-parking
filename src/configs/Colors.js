export const Colors = {
  Primary: '#00979D',
  Text: '#363636',
  Golden: '#b2973f',
  White: '#FFFFFF',
  Red: 'red',
  Green: 'green',
  Gray: '#838383',
  LightGray: '#989898',
  Error: '#F10303',
  DarkLayer: '#00000080',
  LightDarkLayer: '#0000004C',
  LightingDarkLayer: '#00000033',
  LightWhiteLayer: '#F7F7F7D9',
  CoinTagBG: '#F7F7F7',
  Black: '#000000',
  Disabled: '#f8f8f8',
  Border: '#BFBFBF',
  ButtonTag: '#989898',
  WrapGray: '#F5F5F5',
  SwitchButton: '#909090',
  Read: '#F3F3F3',
  Orange: '#F08229',
  BG: '#F0F2F5',
  BGImage: '#F4F8F8',
  Transparent: '#00000000',
  TransparentWhite: '#FFFFFF00',
  BGMapCircle: '#E6F7FF30',
  BGNotRain: '#FEFFE6',
  BorderRain: '#36CFC9',
  Shadow: '#303133',
  ShadownTransparent: 'rgba(48, 49, 51, 0.1)',
  TextTransparent: 'transparent',
  TextGray: 'gray',
  LightOrange: '#FEEFE7',

  //Range Cyan
  Cyan2: '#B5F5EC',
  Cyan2opa30: '#B5F5EC4D',
  Cyan6: '#13C2C2',

  //Range GeekBlue
  GeekBlue3: '#ADC6FF',

  // Range Gray
  Gray4: '#E8E8E8',
  Gray5: '#D9D9D9',
  Gray6: '#BFBFBF',
  Gray7: '#8C8C8C',
  Gray8: '#595959',
  Gray9: '#262626',
  Gray2: '#FAFAFA',
  Gray3: '#F5F5F5',
  Gray11: '#3031331a',
  Gray12: '#DDD',
  Gray13: '#CCC',
  Gray14: '#EEE',
  Gray16: '#333',
  Gray17: '#D3D3D3',
  Gray18: '#9B9A9B',
  Gray19: '#E5E5E5',

  // Range Green
  Green1: '#F6FFED',
  Green3: '#B7EB8F',
  Green6: '#52C41A',
  Green7: '#00979D',
  Green8: '#D9F7BE',
  Green9: '#389E0D',
  Green10: '#237804',

  //Yellow
  Yellow6: '#FADB14',
  Yellow4: '#FFF566',

  //Orange
  Orange3: '#FFD591',
  Orange4: '#FFC069',
  Orange6: '#FA8C16',
  Orange7: '#FAAD14',

  //Red
  Red9: '#A8071A',
  Red8: '#CF1322',
  Red7: '#FA541C',
  Red6: '#F5222D',
  Red5: '#FF4D4F',
  Red4: '#FF7875',
  Red3: '#FFA39E',
  Red2: '#FFCCC7',
  Red1: '#FFF1F0',

  //Purple
  Purple3: '#D3ADF7',
  Purple6: '#722ED1',

  //Pink
  Pink1: '#EFDBFF',

  //Blue
  Blue1: '#F0F5FF',
  Blue2: '#D6E4FF',
  Blue3: '#ADC6FF',
  Blue4: '#85A5FF',
  Blue5: '#597EF7',
  Blue6: '#2F54EB',
  Blue7: '#1D39C4',
  Blue9: '#91D5FF',
  Blue10: '#1890FF',
  Blue11: '#096DD9',
  Blue12: '#0050B3',

  //Range Volcano:
  Volcano3: '#FFBB96',

  //lime
  Lime6: '#A0D911',

  Cyan1: '#E6FFFB',

  LightPrimary: '#EBF7F8',

  BlackTransparent1: 'rgba(0, 0, 0, 0.1)',
  BlackTransparent2: 'rgba(0, 0, 0, 0.2)',
  BlackTransparent3: 'rgba(0, 0, 0, 0.3)',
  BlackTransparent4: 'rgba(0, 0, 0, 0.4)',
  BlackTransparent5: 'rgba(0, 0, 0, 0.5)',
  BlackTransparent6: 'rgba(0, 0, 0, 0.6)',
  BlackTransparent16: 'rgba(0, 0, 0, 0.16)',
  WhiteTransparent8: 'rgba(255, 255, 255, 0.8)',
};
