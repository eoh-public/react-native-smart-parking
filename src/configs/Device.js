/** @format */

import { Dimensions, PixelRatio, Platform } from 'react-native';

const { width, height } = Dimensions.get('window');

const isIphoneX =
  Platform.OS === 'ios' &&
  !Platform.isPad &&
  !Platform.isTVOS &&
  (height >= 812 || width >= 812);

export default {
  pixelRatio: PixelRatio.get(),
  screenWidth: width,
  screenHeight: height,
  isIOS: Platform.OS === 'ios',
  isIphoneX,
  ToolbarHeight: isIphoneX ? 44 : Platform.OS === 'ios' ? 20 : 0,
  TopbarHeight: isIphoneX ? 88 : Platform.OS === 'ios' ? 64 : 44, //incule toolbar
  androidSmallHeight: Platform.OS === 'android' && height < 600,
};
