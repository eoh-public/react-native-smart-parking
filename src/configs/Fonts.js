export const Fonts = {
  Regular: 'SFProDisplay-Regular',
};
