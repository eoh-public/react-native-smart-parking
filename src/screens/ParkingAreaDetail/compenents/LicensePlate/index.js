import React, { memo, useCallback } from 'react';
import { StyleSheet, View, TextInput, TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { t } from 'i18n-js';

import { Colors } from '../../../../configs';
import Text from '../../../../commons/Text';
import { ButtonSaveVehicle } from '../ParkingDetail';
import { SvgCar, SvgCarDisable } from '../../../../../assets/images/SmartParking';

import Routes from '../../../../utils/Route';
import { TESTID } from '../../../../configs/Constants';

const LicensePlate = memo(
  ({
    saveVehicle,
    setSaveVehicle,
    onChangeCar,
    parkingId,
    spot_id,
    spot_name,
    car,
    cars,
    loadingCars,
    refreshCars,
    onRefreshCars,
    numBookHour,
    validCar,
  }) => {
    const { navigate } = useNavigation();
    const onPressCar = useCallback(() => {
      navigate(Routes.SavedVehicle, {
        parkingId,
        spot_id,
        spot_name,
        cars,
        loadingCars,
        refreshCars,
        onRefreshCars,
        numBookHour,
      });
    }, [
      navigate,
      parkingId,
      spot_id,
      spot_name,
      cars,
      loadingCars,
      refreshCars,
      onRefreshCars,
      numBookHour,
    ]);

    return (
      <>
        <Text
          semibold
          style={styles.textLicensePlate}
          color={Colors.Black}
          type="H4"
        >
          {t('license_plate')}
        </Text>
        <View style={validCar ? styles.boxCar : styles.boxInvalidCar}>
          <TextInput
            testID={TESTID.INPUT_PLATE_NUMBER}
            onChangeText={onChangeCar}
            style={styles.numberPlates}
            maxLength={11}
          >
            {car ? car.plate_number : ''}
          </TextInput>
          <TouchableOpacity
            testID={TESTID.PRESS_CAR}
            onPress={onPressCar}
            disabled={saveVehicle}
            style={styles.carButton}
          >
            {!saveVehicle ? (
              <SvgCar height={20} width={20} />
            ) : (
              <SvgCarDisable height={20} width={20} />
            )}
          </TouchableOpacity>
        </View>
        {car && !!car.plate_number && !validCar && (
          <Text style={styles.warning}>
            {t('car_validate_warning', { example: '51A-123.45' })}
          </Text>
        )}
        <ButtonSaveVehicle
          initSave={saveVehicle}
          onChangeSave={setSaveVehicle}
          disabled={!validCar || car.id}
        />
        <View style={styles.line} />
      </>
    );
  }
);

export default LicensePlate;

const styles = StyleSheet.create({
  textLicensePlate: {
    marginLeft: 16,
  },
  boxCar: {
    flexDirection: 'row',
    height: 40,
    alignItems: 'center',
    marginHorizontal: 16,
    paddingLeft: 12,
    backgroundColor: Colors.White,
    marginTop: 13,
    borderRadius: 2,
    borderWidth: 1,
    borderColor: Colors.Gray5,
    overflow: 'hidden',
  },
  boxInvalidCar: {
    flexDirection: 'row',
    height: 40,
    alignItems: 'center',
    marginHorizontal: 16,
    paddingLeft: 12,
    backgroundColor: Colors.White,
    marginTop: 13,
    borderRadius: 2,
    borderWidth: 1,
    borderColor: Colors.Red,
    overflow: 'hidden',
  },
  numberPlates: {
    flex: 1,
  },
  carButton: {
    paddingVertical: '100%',
    paddingHorizontal: 14,
  },
  line: {
    height: 1,
    backgroundColor: Colors.Gray4,
    marginVertical: 16,
  },
  warning: {
    marginTop: 5,
    marginLeft: 20,
    color: Colors.Red,
  },
});
