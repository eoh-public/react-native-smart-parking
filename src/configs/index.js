import { SPConfig, initSPConfig } from './SPConfig';
import API from './API';
import { Colors } from './Colors';
import { Constants } from './Constants';
import Theme from './Theme';
import Images from './Images';
import Styles from './Styles';
import BLE from './BLE';
import Device from './Device';
import { Fonts } from './Fonts';
export {
  API,
  BLE,
  Colors,
  Constants,
  Theme,
  Styles,
  Images,
  Device,
  Fonts,
  SPConfig,
  initSPConfig,
};
