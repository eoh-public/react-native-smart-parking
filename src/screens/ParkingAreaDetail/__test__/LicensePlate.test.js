import React from 'react';
import { TouchableOpacity } from 'react-native';
import { create, act } from 'react-test-renderer';
import LicensePlate from '../compenents/LicensePlate';
import { TESTID } from '../../../configs/Constants';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe('Test LicensePlate', () => {
  let data;
  const mockSetSaveVehicle = jest.fn();
  const mockOnChangeCar = jest.fn();
  const mockRefreshCars = jest.fn();

  beforeEach(() => {
    data = {
      saveVehicle: false,
      setSaveVehicle: mockSetSaveVehicle,
      onChangeCar: mockOnChangeCar,
      parkingId: 1,
      spot_id: 1,
      spot_name: '',
      car: {
        plate_number: '59Z - 1234',
        id: 1,
      },
      cars: [{ id: 1, seats: 1, name: 'Merc', plate_number: '59Z - 1234' }],
      loadingCars: false,
      refreshCars: mockRefreshCars,
      onRefreshCars: mockRefreshCars,
      numBookHour: 1,
      validCar: false,
    };
  });
  let wrapper;

  test('create LicensePlate', () => {
    act(() => {
      wrapper = create(<LicensePlate {...data} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('create LicensePlate with validCar', () => {
    data.validCar = true;
    act(() => {
      wrapper = create(<LicensePlate {...data} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('create LicensePlate with savedVehicle', () => {
    // TODO distinguish two svg
    data.savedVehicle = true;
    act(() => {
      wrapper = create(<LicensePlate {...data} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('create LicensePlate without car', () => {
    data.car = null;
    act(() => {
      wrapper = create(<LicensePlate {...data} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('onPressItem LicensePlate', () => {
    act(() => {
      wrapper = create(<LicensePlate {...data} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
    const instance = wrapper.root;

    const buttonOnPressCar = instance.find(
      (el) =>
        el.props.testID === TESTID.PRESS_CAR && el.type === TouchableOpacity
    );

    expect(buttonOnPressCar).not.toBeUndefined();

    act(() => {
      buttonOnPressCar.props.onPress();
    });

    expect(mockedNavigate).toHaveBeenCalled();
  });
});
