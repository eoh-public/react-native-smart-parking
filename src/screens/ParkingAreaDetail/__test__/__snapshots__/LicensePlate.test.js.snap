// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LicensePlate create LicensePlate 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 16,
        },
      ]
    }
  >
    Biển số xe
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "red",
        "borderRadius": 2,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginHorizontal": 16,
        "marginTop": 13,
        "overflow": "hidden",
        "paddingLeft": 12,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      maxLength={11}
      onChangeText={[MockFunction]}
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="INPUT_PLATE_NUMBER"
      underlineColorAndroid="transparent"
    >
      59Z - 1234
    </TextInput>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 14,
          "paddingVertical": "100%",
        }
      }
      testID="PRESS_CAR"
    >
      <SvgMock
        height={20}
        width={20}
      />
    </View>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": undefined,
          "lineHeight": undefined,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "color": "red",
          "marginLeft": 20,
          "marginTop": 5,
        },
      ]
    }
  >
    Vui lòng nhập đúng định dạng biển số xe 
VD: 50A-123.45
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      pointerEvents="none"
    >
      <RNCCheckbox
        boxType="square"
        forwardedRef={null}
        onCheckColor="#FFFFFF"
        onFillColor="#00979D"
        onTintColor="#00979D"
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            Array [
              Object {
                "height": 20,
                "marginRight": 15,
                "marginTop": 15,
                "width": 20,
              },
              undefined,
            ],
          ]
        }
        tintColors={
          Object {
            "false": "#F5F5F5",
            "true": "#00979D",
          }
        }
        value={false}
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#BFBFBF",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginTop": 14,
            },
          ]
        }
      >
        Lưu và đặt phương tiện mặc định
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test LicensePlate create LicensePlate with savedVehicle 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 16,
        },
      ]
    }
  >
    Biển số xe
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "red",
        "borderRadius": 2,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginHorizontal": 16,
        "marginTop": 13,
        "overflow": "hidden",
        "paddingLeft": 12,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      maxLength={11}
      onChangeText={[MockFunction]}
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="INPUT_PLATE_NUMBER"
      underlineColorAndroid="transparent"
    >
      59Z - 1234
    </TextInput>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 14,
          "paddingVertical": "100%",
        }
      }
      testID="PRESS_CAR"
    >
      <SvgMock
        height={20}
        width={20}
      />
    </View>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": undefined,
          "lineHeight": undefined,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "color": "red",
          "marginLeft": 20,
          "marginTop": 5,
        },
      ]
    }
  >
    Vui lòng nhập đúng định dạng biển số xe 
VD: 50A-123.45
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      pointerEvents="none"
    >
      <RNCCheckbox
        boxType="square"
        forwardedRef={null}
        onCheckColor="#FFFFFF"
        onFillColor="#00979D"
        onTintColor="#00979D"
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            Array [
              Object {
                "height": 20,
                "marginRight": 15,
                "marginTop": 15,
                "width": 20,
              },
              undefined,
            ],
          ]
        }
        tintColors={
          Object {
            "false": "#F5F5F5",
            "true": "#00979D",
          }
        }
        value={false}
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#BFBFBF",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginTop": 14,
            },
          ]
        }
      >
        Lưu và đặt phương tiện mặc định
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test LicensePlate create LicensePlate with validCar 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 16,
        },
      ]
    }
  >
    Biển số xe
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#D9D9D9",
        "borderRadius": 2,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginHorizontal": 16,
        "marginTop": 13,
        "overflow": "hidden",
        "paddingLeft": 12,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      maxLength={11}
      onChangeText={[MockFunction]}
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="INPUT_PLATE_NUMBER"
      underlineColorAndroid="transparent"
    >
      59Z - 1234
    </TextInput>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 14,
          "paddingVertical": "100%",
        }
      }
      testID="PRESS_CAR"
    >
      <SvgMock
        height={20}
        width={20}
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      pointerEvents="none"
    >
      <RNCCheckbox
        boxType="square"
        forwardedRef={null}
        onCheckColor="#FFFFFF"
        onFillColor="#00979D"
        onTintColor="#00979D"
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            Array [
              Object {
                "height": 20,
                "marginRight": 15,
                "marginTop": 15,
                "width": 20,
              },
              undefined,
            ],
          ]
        }
        tintColors={
          Object {
            "false": "#F5F5F5",
            "true": "#00979D",
          }
        }
        value={false}
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#BFBFBF",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginTop": 14,
            },
          ]
        }
      >
        Lưu và đặt phương tiện mặc định
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test LicensePlate create LicensePlate without car 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 16,
        },
      ]
    }
  >
    Biển số xe
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "red",
        "borderRadius": 2,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginHorizontal": 16,
        "marginTop": 13,
        "overflow": "hidden",
        "paddingLeft": 12,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      maxLength={11}
      onChangeText={[MockFunction]}
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="INPUT_PLATE_NUMBER"
      underlineColorAndroid="transparent"
    >
      
    </TextInput>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 14,
          "paddingVertical": "100%",
        }
      }
      testID="PRESS_CAR"
    >
      <SvgMock
        height={20}
        width={20}
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      pointerEvents="none"
    >
      <RNCCheckbox
        boxType="square"
        forwardedRef={null}
        onCheckColor="#FFFFFF"
        onFillColor="#00979D"
        onTintColor="#00979D"
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            Array [
              Object {
                "height": 20,
                "marginRight": 15,
                "marginTop": 15,
                "width": 20,
              },
              undefined,
            ],
          ]
        }
        tintColors={
          Object {
            "false": "#F5F5F5",
            "true": "#00979D",
          }
        }
        value={false}
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#BFBFBF",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginTop": 14,
            },
          ]
        }
      >
        Lưu và đặt phương tiện mặc định
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test LicensePlate onPressItem LicensePlate 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginLeft": 16,
        },
      ]
    }
  >
    Biển số xe
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "red",
        "borderRadius": 2,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginHorizontal": 16,
        "marginTop": 13,
        "overflow": "hidden",
        "paddingLeft": 12,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      maxLength={11}
      onChangeText={[MockFunction]}
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
        }
      }
      testID="INPUT_PLATE_NUMBER"
      underlineColorAndroid="transparent"
    >
      59Z - 1234
    </TextInput>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 14,
          "paddingVertical": "100%",
        }
      }
      testID="PRESS_CAR"
    >
      <SvgMock
        height={20}
        width={20}
      />
    </View>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": undefined,
          "lineHeight": undefined,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "color": "red",
          "marginLeft": 20,
          "marginTop": 5,
        },
      ]
    }
  >
    Vui lòng nhập đúng định dạng biển số xe 
VD: 50A-123.45
  </Text>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      pointerEvents="none"
    >
      <RNCCheckbox
        boxType="square"
        forwardedRef={null}
        onCheckColor="#FFFFFF"
        onFillColor="#00979D"
        onTintColor="#00979D"
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            Array [
              Object {
                "height": 20,
                "marginRight": 15,
                "marginTop": 15,
                "width": 20,
              },
              undefined,
            ],
          ]
        }
        tintColors={
          Object {
            "false": "#F5F5F5",
            "true": "#00979D",
          }
        }
        value={false}
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#BFBFBF",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginTop": 14,
            },
          ]
        }
      >
        Lưu và đặt phương tiện mặc định
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;
